apps/indexer/src/api/services/proposals/index.ts

```diff
@@ -81,28 +83,43 @@ export class ProposalsService {
     status,
     fromDate,
     fromEndDate,
+    includeOptimisticProposals = true,
   }: ProposalsRequest): Promise<DBProposal[]> {
     // 1. Prepare status for database query
     const dbStatuses = status
       ? this.prepareStatusForDatabase(status)
       : undefined;
 
-    // 2. Fetch proposals from database
+    // 2.Filter proposal type using DAO config
+    let proposalTypeExclude: number[] | undefined;
+    if (!includeOptimisticProposals) {
+      const daoConfig = CONTRACT_ADDRESSES[env.DAO_ID];
+      const optimisticType =
+        "optimisticProposalType" in daoConfig
+          ? daoConfig.optimisticProposalType
+          : undefined;
+      if (optimisticType !== undefined) {
+        proposalTypeExclude = [optimisticType];
+      }
+    }
```

Comment:
could that be injected? having a direct dependecy on the the env variables makes it harder to test and understand this code